src/ecmascript/spidermonkey/document.c

```diff
@@ -492,3 +552,122 @@ document_writeln(JSContext *ctx, unsigned int argc, JS::Value *rval)
 {
 	return document_write_do(ctx, argc, rval, 1);
 }
+
+// Helper function for document replace
+char *
+str_replace(char *orig, char *rep, char *with)
+{
+	char *result; // the return string
+	char *ins;    // the next insert point
+	char *tmp;    // varies
+	int len_rep;  // length of rep (the string to remove)
+	int len_with; // length of with (the string to replace rep with)
+	int len_front; // distance between rep and end of last rep
+	int count;    // number of replacements
+
+	// sanity checks and initialization
+	if (!orig || !rep)
+	{
+		return NULL;
+	}
+	len_rep = strlen(rep);
+	if (len_rep == 0) {
+		return NULL; // empty rep causes infinite loop during count
+	}
+	if (!with) {
+		with = "";
+	}
+	len_with = strlen(with);
+
+	// count the number of replacements needed
+	ins = orig;
+	for (count = 0; tmp = strstr(ins, rep); ++count)
+	{
+		ins = tmp + len_rep;
+	}
+
+	result = tmp = malloc(strlen(orig) + ( (len_with - len_rep) * count) + 1);
+
+	if (!result) {
+		return NULL;
+	}
+
+	// first time through the loop, all the variable are set correctly
+	// from here on,
+	//    tmp points to the end of the result string
+	//    ins points to the next occurrence of rep in orig
+	//    orig points to the remainder of orig after "end of rep"
+	while (count--)
+	{
+		ins = strstr(orig, rep);
+		len_front = ins - orig;
+		tmp = strncpy(tmp, orig, len_front) + len_front;
+		tmp = strcpy(tmp, with) + len_with;
+		orig += len_front + len_rep; // move to next "end of rep"
+	}
+	strcpy(tmp, orig);
+	//sprintf(tmp,'%s',orig);
+	return(result);
+}
+
+
+/* @document_funcs{"replace"} */
+static bool
+document_replace(JSContext *ctx, unsigned int argc, JS::Value *vp)
+{
+
+	JSCompartment *comp = js::GetContextCompartment(ctx);
+	struct ecmascript_interpreter *interpreter = JS_GetCompartmentPrivate(comp);
+	struct document_view *doc_view = interpreter->vs->doc_view;
+	struct session *ses = doc_view->session;
+	struct terminal *term = ses->tab->term;
+	struct string *ret = interpreter->ret;
+	struct document *document;
+	JS::CallArgs args = CallArgsFromVp(argc, vp);
+	document = doc_view->document;
+
+	if (argc != 2) {
+		args.rval().setBoolean(false);
+		return(true);
+	}
+
+	unsigned char *needle;
+	unsigned char *heystack;
+
+	needle = stracpy("");
+	heystack = stracpy("");
+
+
+	needle = jshandle_value_to_char_string(ctx,&args[0]);
+	heystack = jshandle_value_to_char_string(ctx,&args[1]);
+
+	//DBG("doc replace %s %s\n", needle, heystack);
+
+	int nu_len=0;
+	int fd_len=0;
+	unsigned char *nu;
+	struct cache_entry *cached = doc_view->document->cached;
+	struct fragment *f = cached ? cached->frag.next : NULL;
```

Comment:
assigmnment to f not needed, because below f is assigned once again.

Reply:
O.K., agreed. Fixed.